src/Microsoft.Performance.SDK/Processing/ITableDiscoverer.cs

```diff
@@ -0,0 +1,309 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Performance.SDK.Extensibility;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Provides an interface for providing tables that are exposed
+    ///     by <see cref="ProcessingSource"/>s.
+    /// </summary>
+    public interface ITableProvider
+    {
+        /// <summary>
+        ///     Returns a collection of tables.
```

Comment:
nit: this is a vague description. Maybe a "Returns the collection of tables that should be associated with a ProcessingSource" or something? Any thoughts?

Reply:
That sounds better, I will update.